lib/mocha.js

```diff
@@ -424,23 +455,45 @@ Mocha.prototype.run = function(fn){
   var options = this.options;
   options.files = this.files;
   var runner = new exports.Runner(suite, options.delay);
-  var reporter = new this._reporter(runner, options);
+  // For each loaded reporter constructor, create
+  // an instance and initialize it with the runner
+  var reporters = this._reporters.map(function (reporterConfig) {
+    var _reporter = reporterConfig.fn
+      , path = reporterConfig.path;
+
+    options.reporterOptions = reporterConfig.options;
+    return new _reporter(runner, options, path);
+  });
   runner.ignoreLeaks = false !== options.ignoreLeaks;
   runner.fullStackTrace = options.fullStackTrace;
   runner.asyncOnly = options.asyncOnly;
   if (options.grep) runner.grep(options.grep, options.invert);
   if (options.globals) runner.globals(options.globals);
-  if (options.growl) this._growl(runner, reporter);
+
+  // Use only the first reporter for growl, since we don't want to
+  // send several notifications for the same test suite
+  if (options.growl) this._growl(runner, reporters[0]);
+
   if (options.useColors !== undefined) {
     exports.reporters.Base.useColors = options.useColors;
   }
+
   exports.reporters.Base.inlineDiffs = options.useInlineDiffs;
 
-  function done(failures) {
+  function runnerDone(failures) {
+    var remain = reporters.length
+      , reporterDone = function(failures) {
```

Comment:
fyi: with the linting work that's being done, we're not going to be using leading commas.  also, the below conditional will need curlies